tests/unit/dataactvalidator/test_d39_detached_award_financial_assistance_3.py

```diff
@@ -0,0 +1,40 @@
+from tests.unit.dataactcore.factories.staging import DetachedAwardFinancialAssistanceFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+
+_FILE = 'd39_detached_award_financial_assistance_3'
+
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "place_of_performance_code", "place_of_performance_zip4a"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset == actual
+
+
+def test_success(database):
+    """ PrimaryPlaceofPerformanceZIP+4 should not be provided for any format of PrimaryPlaceOfPerformanceCode
+        other than XX#####. """
+
+    det_award_1 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="NY12345",
+                                                          place_of_performance_zip4a="1234")
+    det_award_2 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code="ny98765",
+                                                          place_of_performance_zip4a="4312")
+    det_award_3 = DetachedAwardFinancialAssistanceFactory(place_of_performance_code=None,
```

Comment:
We probably actually want it to pass but for a different reason. It SHOULD fail if it's None but `place_of_performance_code` is a required field which means we'd have 2 errors showing up for the same problem if it was both missing and throwing an error for `None` as well. What's your opinion on how we should handle this conundrum?